src/application/validation/validators/notification-mobile-validation.ts

```diff
@@ -0,0 +1,23 @@
+import { type Validation } from '@/application/protocols'
+import { type BooleanValidator } from '../protocols'
+import { InvalidParamError } from '@/application/errors'
+
+export class NotificationMobileValidation implements Validation {
+  private readonly fieldName: string
+  private readonly validator: BooleanValidator
+
+  constructor (fieldName: string, validator: BooleanValidator) {
+    this.fieldName = fieldName
+    this.validator = validator
+  }
+
+  validate (input: any): Error | void {
+    if (typeof input[this.fieldName] !== 'boolean' && input[this.fieldName] !== null) {
```

Comment:
Acredito que não precise da segunda condição, já que a primeira condição só passa se o tipo for um booleano (true ou false) se for undefined, null ou outra coisa essa primeira condição já impediria de passar para o resto do código

```suggestion
    if (typeof input[this.fieldName] !== 'boolean') {
```